app/controllers/api/v1/current_user_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Api
+  module V1
+    class CurrentUserController < ApiBaseController
+      # collection of API methods related to the current user object.
+      # for security reasons, the methods in this controller should be extremely tightly controlled
+      include Concerns::Authenticator
+      include Concerns::StudyAware
+      include Swagger::Blocks
+
+      before_action :set_current_api_user!
+
+      ALLOWABLE_UPDATE_FIELDS = ['feature_flags']
+      # to ensure users can't arbitrarily change their own flags, restrict the list
+      ALLOWABLE_UPDATE_FEATURE_FLAGS = ['faceted_search']
+      # updates the current user object -- only allowed fields are able to be changed
+      def update
+        user = current_api_user
+
+        if user.nil?
+          # anonymous users can't change feature flags
+          head 403 and return
+        end
+
+        updated_user = params[:current_user]
+        # if there are any fields included that are not allowed to be updated, return error
+        if (ALLOWABLE_UPDATE_FIELDS - updated_user.keys).present?
```

Comment:
I think that there's a cleaner way of doing this.  It seems to me that since you're trying to save values, but want validations to be enforced, we should be specifying those at the model level (or in the `feature_flaggable` concern), and then all you need to do is check if the `user` model saved.